File: src/main/java/rk/cinema/repository/CustomerRepository.java
package rk.cinema.repository;

import io.micronaut.core.annotation.NonNull;
import io.micronaut.data.jdbc.annotation.JdbcRepository;
import io.micronaut.data.model.query.builder.sql.Dialect;
import io.micronaut.data.repository.CrudRepository;
import rk.cinema.model.Customer;

import java.util.Optional;

@JdbcRepository(dialect = Dialect.POSTGRES)
public interface CustomerRepository extends CrudRepository<Customer, Long> {
    Optional<Customer> findByEmail(@NonNull String email);
}

File: src/main/kotlin/rk/powermilk/cinema/repository/CustomerRepository.kt
package rk.powermilk.cinema.repository

import io.micronaut.context.annotation.Executable
import io.micronaut.core.annotation.NonNull
import io.micronaut.data.jdbc.annotation.JdbcRepository
import io.micronaut.data.model.query.builder.sql.Dialect
import io.micronaut.data.repository.CrudRepository
import rk.powermilk.cinema.model.Customer

@JdbcRepository(dialect = Dialect.POSTGRES)
interface CustomerRepository : CrudRepository<Customer, Long> {
    @Executable
    fun findByEmail(email: @NonNull String): Customer
}
